Rubberduck.CodeAnalysis/Inspections/Concrete/AssignmentNotUsedInspection.cs

```diff
@@ -98,7 +100,8 @@ public static List<IdentifierReference> UnusedAssignmentReferences(INode node)
 
         protected override bool IsResultReference(IdentifierReference reference, DeclarationFinder finder)
         {
-            return !IsAssignmentOfNothing(reference);
+            return !(IsAssignmentOfNothing(reference)
+                        || IsPotentiallyUsedViaResumeOrGoToExecutionBranch(reference, finder));
```

Comment:
That name feels a bit long-winded... how about `IsPotentiallyUsedViaJump`?

Reply:
Agreed - changed